cosmos/providers/dbt/render.py

```diff
@@ -23,18 +26,19 @@ def render_project(
     dbt_project_name: str,
     dbt_root_path: str = "/usr/local/airflow/dbt",
     dbt_models_dir: str = "models",
-    task_args: Dict[str, Any] = {},
+    task_args: DbtArgs | None = None,
     test_behavior: Literal["none", "after_each", "after_all"] = "after_each",
     emit_datasets: bool = True,
     conn_id: str = "default_conn_id",
-    select: Dict[str, List[str]] = {},
-    exclude: Dict[str, List[str]] = {},
+    select: dict[str, list[str]] | None = None,
+    exclude: dict[str, list[str]] | None = None,
```

Comment:
just to confirm - does this work in python 3.7 because of the `from __future__ import annotations`? we had a PR a while ago that seemed to make these changes despite the import

https://github.com/astronomer/astronomer-cosmos/pull/88

Reply:
As I understand it from Python 3.7 they made this annotations future available which implements PEP 563.
https://docs.python.org/3.7/library/__future__.html

I dug out the PR on CPython and noticed a comment from Guido here https://github.com/python/cpython/pull/4390#issuecomment-344116700
Although it must have been called something different originally you can see him checking by using the built-in types.